src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -1,48 +1,83 @@
-
 #include <iostream>
 #include <ctime>
 
 #include "MUSI6106Config.h"
 
 #include "AudioFileIf.h"
+#include "CombFilterIf.h"
 
 using std::cout;
 using std::endl;
 
 // local function declarations
 void    showClInfo ();
 
+void example1 ();
+void example2 ();
+void test1 ();
+void test2 ();
+void test3_FIR ();
+void test3_IIR ();
+void test4_FIR ();
+void test4_IIR ();
+void test5();
+
+typedef struct arg{
+    std::string inputAudioPath;
+    std::string outputAudioPath;
+    int blockSize;
+    CCombFilterIf::CombFilterType_t filterType;
+    float gain;
+    float delay;};
+
 /////////////////////////////////////////////////////////////////////////////////
 // main function
 int main(int argc, char* argv[])
 {
-    std::string             sInputFilePath,                 //!< file paths
-                            sOutputFilePath;
+    example1();
+    cout << "Example 1" << endl;
+    example2();
+    cout << "Example 2" << endl;
+    test1 ();
+    cout << "Test 1" << endl;
+    test2 ();
+    cout << "Test 2" << endl;
+    test3_FIR ();
+    cout << "Test 3 FIR" << endl;
+    test3_IIR ();
+    cout << "Test 3 IIR" << endl;
+    test4_FIR ();
+    cout << "Test 4 FIR" << endl;
+    test4_IIR ();
+    cout << "Test 4 IIR" << endl;
+    test5 ();
+    cout << "Test 5" << endl;
+    showClInfo();
+    return 0;
+}
+
+int filtering(arg argall){
```

Comment:
I think the structure is really great! I should also try this. From user's point of view. It might be also to have the option run certain test only.